### PR TITLE
feat(directive): add decomposition readiness layer

### PR DESCRIPTION
**Pull request by dbcall2**
## Summary

Adds a first-class Directive story decomposition/readiness layer between Speckit Phase 4 and swarm implementation.

Highlights:
- Documents Phase 4.5 Story Decomposition / Swarm Readiness.
- Adds deterministic `task scope:decompose` and `task swarm:readiness` commands.
- Defines the canonical `plan.metadata.kind` / `plan.metadata.swarm` story contract.
- Updates swarm allocation to require readiness reports before worker dispatch.
- Adds `deft-directive-decompose` skill and fixes Speckit/template v0.6 reference patterns.
- Preserves epic acceptance and story acceptance in spec rendering.
- Rebases onto current `master` and aligns generated in-repo references with the new `TrustLevel` provenance contract.
- Tightens the story-quality contract: product-shaped `UserStory`, concrete observable acceptance, narrow file scope, focused verification, and `ready` meaning concurrent-ready only.
- Requires substantive story plans: each ready story needs concrete `Description` and `ImplementationPlan` narratives, not just structurally complete metadata.
- Addresses review feedback for sequential-only stories, cycle reporting, semicolon acceptance rendering, shared TrustLevel defaulting, shared Speckit translation, required swarm metadata enforcement, blocked dependency propagation, explicit story kind enforcement, parent-shape validation, child provenance filtering, vBRIEF-relative URI/planRef emission, intentional Taskfile CLI arg pass-through, and required swarm Taskfile inclusion.

## Related Issues

N/A - user-requested framework change in this session.

## Checklist

- [x] `/deft:change <name>` - scope vBRIEF created, activated, and completed for this work
- [x] `CHANGELOG.md` - added entry under `[Unreleased]`
- [x] `ROADMAP.md` - N/A, no tracked issue state changed
- [x] Tests pass locally: `task check` -> passed

## Verification

- `uv run ruff check scripts/scope_decompose.py scripts/_vbrief_speckit.py scripts/migrate_vbrief.py tests/cli/test_scope_decompose.py` -> passed
- `uv run pytest tests/cli/test_scope_decompose.py tests/cli/test_migrate_vbrief.py tests/content/test_speckit_phase4.py tests/content/test_decomposition_readiness.py -q` -> 239 passed
- `uv run pytest tests/cli/test_scope_decompose.py tests/cli/test_swarm_readiness.py tests/cli/test_migrate_vbrief.py tests/cli/test_spec_render.py tests/content/test_decomposition_readiness.py tests/content/test_speckit_phase4.py tests/content/test_swarm_skill.py tests/content/test_taskfile_cli_args.py tests/content/test_taskfile_uv_project_pin.py tests/content/test_taskfile_paths.py -q` -> 403 passed, 2 deselected
- `task --list-all | rg "swarm:readiness|scope:decompose"` -> both targets resolve
- `task check` -> passed
- `git diff --check` -> clean

## Post-Merge

- [ ] N/A for issue auto-close; no GitHub issue is linked by this PR body.
- [ ] Enable branch protection on `master` requiring CI status check (one-time setup, see #57)

### CODE REVIEW

**Comment by dbcall2** [comment]
Review feedback addressed in `621d2df`:

- `parallel_safe=false` is now treated as an explicit sequential-only declaration, not a missing field; readiness reports include a `Sequential stories` section and fail concurrent readiness when candidates require sequential allocation.
- Dependency cycle reporting now slices from the repeated node, so upstream non-cycle stories are not labeled as cyclic.
- Acceptance rendering no longer splits semicolon text; explicit list values or bullet-prefixed lines remain the way to render multiple criteria.
- TrustLevel defaulting is centralized in `_vbrief_build.reference_with_default_trust`, and Speckit/decompose/migrate paths reuse it with consistent deep-copy semantics.

Verification after the fixes:
- `uv run pytest tests/cli/test_scope_decompose.py tests/cli/test_swarm_readiness.py tests/cli/test_migrate_vbrief.py tests/cli/test_spec_render.py tests/content/test_decomposition_readiness.py tests/content/test_speckit_phase4.py tests/content/test_swarm_skill.py -q` -> 262 passed
- `uv run pytest tests/cli/test_spec.py tests/cli/test_spec_render.py -q` -> 51 passed
- `task check` -> 5149 passed, 5 skipped, 8 deselected, 1 xfailed
- `git diff --check` -> clean

There are no formal GitHub review threads to resolve on this PR (`reviewThreads` is empty); the prior bot feedback was posted as top-level comments/reviews.

**Comment by dbcall2** [comment]
Follow-up review fixes are pushed through `cd02744`.

What changed after the latest SLizard thread:
- `task swarm:readiness` now reports missing `swarm.expected_outputs`, explicit `swarm.depends_on`, `swarm.conflict_group`, `swarm.size`, `swarm.file_scope_confidence`, and `swarm.model_tier` for `readiness: ready` stories.
- `task scope:decompose` now applies the same required swarm metadata contract before writing child story vBRIEFs.
- Added regression coverage for missing required swarm metadata.
- Also cleaned up the prior Greptile follow-ups: decompose DAG cycle messages now omit upstream non-cycle stories, readiness overlap checking no longer has the unreachable `parallel_safe` guard, and `migrate_vbrief.py` delegates Speckit translation to the shared `_vbrief_speckit.py` implementation.

Verification:
- `uv run ruff check scripts/scope_decompose.py scripts/swarm_readiness.py tests/cli/test_swarm_readiness.py tests/cli/test_scope_decompose.py` -> passed
- `uv run pytest tests/cli/test_scope_decompose.py tests/cli/test_swarm_readiness.py tests/cli/test_migrate_vbrief.py tests/cli/test_spec_render.py tests/content/test_decomposition_readiness.py tests/content/test_speckit_phase4.py tests/content/test_swarm_skill.py -q` -> 264 passed
- `git diff --check` -> clean

Current PR status: Greptile passed, Socket checks passed, SLizard completed neutral/skipped on latest head, and the review thread has been resolved.

**Comment by dbcall2** [comment]
Implemented the tighter story-quality contract in `e7ead24`.

What changed:
- Added a shared story-quality gate used by both `scope_decompose.py` and `swarm_readiness.py`.
- `deft-directive-decompose` now requires product-shaped `UserStory` text, 2-5 observable acceptance criteria unless justified, and treats parent `plan.items` only as input signals.
- Ready stories are rejected for placeholder acceptance, criteria that duplicate title/description, vague docs-only acceptance, broad file globs like `backend/**`, `frontend/**`, `docs/**`, `vbrief/**`, and generic verification like only `task check`.
- `ready` now means concurrent-ready: `parallel_safe=false` and `file_scope_confidence=low` cannot pass as ready; non-concurrent work should use `sequential` or `needs_refinement` and will not pass swarm readiness.
- Updated `vbrief/vbrief.md`, `strategies/speckit.md`, and the swarm/decompose skills to match the state model.

Verification:
- `uv run ruff check scripts/_vbrief_story_quality.py scripts/scope_decompose.py scripts/swarm_readiness.py tests/cli/test_scope_decompose.py tests/cli/test_swarm_readiness.py tests/content/test_decomposition_readiness.py` -> passed
- `uv run pytest tests/cli/test_scope_decompose.py tests/cli/test_swarm_readiness.py tests/content/test_decomposition_readiness.py -q` -> 32 passed
- `uv run pytest tests/cli/test_scope_decompose.py tests/cli/test_swarm_readiness.py tests/cli/test_migrate_vbrief.py tests/cli/test_spec_render.py tests/content/test_decomposition_readiness.py tests/content/test_speckit_phase4.py tests/content/test_swarm_skill.py -q` -> 277 passed
- `git diff --check` -> clean

Current checks on the PR: Greptile passed, Socket passed, SLizard completed neutral/skipping. No unresolved review threads remain.

**Comment by dbcall2** [comment]
Final local verification after the story-contract follow-up:

- `task check` -> 5164 passed, 5 skipped, 8 deselected, 1 xfailed

I temporarily moved the unrelated untracked local `deft/` directory aside for that full check and restored it afterward.

**Comment by dbcall2** [comment]
Addressed the latest SLizard findings in `39208d4`.

What changed:
- `scope_decompose._load_json` now wraps `OSError` as `DecompositionError` with exception chaining, so unreadable JSON files produce clean CLI errors instead of tracebacks.
- Moved duplicated story-contract helpers (`items_have_acceptance`, `item_has_traces`, required swarm metadata checks, `as_str_list` use) into `_vbrief_story_quality.py` and reused them from both decomposition and readiness.
- Tightened ready-story validation: rejects generic/vague implementation plans, vague observable-looking acceptance such as “The system displays a message,” any globbed file scope such as `src/*.ts`, and deprecated `subItems` in story items.
- Added regressions for the `_vbrief_build` TrustLevel fallback concern, indented acceptance bullet rendering, unreadable decomposition JSON, broad glob scopes, deprecated `subItems`, and low-quality story text.
- Clarified the completed change vBRIEF acceptance evidence so it points at the exact skills/docs/tests that support the claim.

Verification:
- `uv run ruff check scripts/_vbrief_story_quality.py scripts/scope_decompose.py scripts/swarm_readiness.py scripts/spec_render.py tests/cli/test_scope_decompose.py tests/cli/test_swarm_readiness.py tests/cli/test_spec_render.py tests/cli/test_migrate_vbrief.py` -> passed
- `uv run pytest tests/cli/test_scope_decompose.py tests/cli/test_swarm_readiness.py tests/cli/test_spec_render.py tests/cli/test_migrate_vbrief.py -q` -> 256 passed
- `uv run pytest tests/cli/test_scope_decompose.py tests/cli/test_swarm_readiness.py tests/cli/test_migrate_vbrief.py tests/cli/test_spec_render.py tests/content/test_decomposition_readiness.py tests/content/test_speckit_phase4.py tests/content/test_swarm_skill.py tests/content/test_taskfile_cli_args.py tests/content/test_taskfile_uv_project_pin.py tests/content/test_taskfile_paths.py -q` -> 414 passed, 2 deselected
- `task check` -> 5184 passed, 5 skipped, 8 deselected, 1 xfailed
- `git diff --check` -> clean

No unresolved GitHub review threads remain. Socket checks are green on the new head; SLizard and Greptile are still pending as of this comment.

File: scripts/migrate_vbrief.py [hunk 1]
```diff
@@ -2277,164 +2243,30 @@ def _create_speckit_scope_vbrief(
...
-    description = _pick("Description", "Summary", default=title)
-    acceptance = _pick("Acceptance", "AcceptanceCriteria", default="")
-    traces = _pick("Traces", "Trace", "Requirements", default=f"IP-{ip_index}")
-
-    default_acceptance = (
-        f"Acceptance criteria for IP-{ip_index} "
-        f"(copy from specification.vbrief.json)."
+    """Compatibility shim for the shared Speckit translator."""
+    return _create_speckit_scope_vbrief_shared(
+        item,
```
[the remark this answers is not in the document]
Comment (dbcall2) [reply scripts/migrate_vbrief.py:2248]: Fixed in `c1ffc19`.

- Removed the stale `../specification.vbrief.json` shape from Speckit translator tests.
- `migrate_speckit_plan` continues to default to `specification.vbrief.json`, and the end-to-end test now asserts that exact vbrief-relative URI.
- Added a defensive normalization in `_vbrief_speckit.create_speckit_scope_vbrief` so legacy parent-traversal input cannot leak into emitted `plan.references[].uri`.
- Verified no `../specification.vbrief.json` occurrences remain under scripts/tests/docs with `rg`.

Also fixed Greptile’s adjacent `output_dir: null` edge case in `scope_decompose.py` and added regression coverage.
